pkg/cli/clisqlshell/editor_bufio.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package clisqlshell
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/cockroachdb/errors"
+)
+
+// bufioReader implements the editor interface.
```

Comment:
could we include declarations for the implementing types that assert that they implement the interface?

Reply:
Good idea. Done.